capsules/src/i2c_master_slave_driver.rs

```diff
@@ -222,73 +237,133 @@ impl hil::i2c::I2CHwSlaveClient for I2CMasterSlaveDriver<'_> {
 impl Driver for I2CMasterSlaveDriver<'_> {
     fn allow_readwrite(
         &self,
-        _appid: ProcessId,
+        process_id: ProcessId,
         allow_num: usize,
         mut slice: ReadWriteAppSlice,
     ) -> Result<ReadWriteAppSlice, (ReadWriteAppSlice, ErrorCode)> {
-        match allow_num {
-            // Pass in a buffer for transmitting a `write` to another
-            // I2C device.
-            0 => {
-                self.app.map(|app| {
-                    core::mem::swap(&mut app.master_tx_buffer, &mut slice);
-                });
-                Ok(slice)
-            }
-            // Pass in a buffer for doing a read from another I2C device.
-            1 => {
-                self.app.map(|app| {
-                    core::mem::swap(&mut app.master_rx_buffer, &mut slice);
-                });
-                Ok(slice)
-            }
-            // Pass in a buffer for handling a read issued by another I2C master.
-            2 => {
-                self.app.map(|app| {
-                    core::mem::swap(&mut app.slave_tx_buffer, &mut slice);
-                });
-                Ok(slice)
-            }
-            // Pass in a buffer for handling a write issued by another I2C master.
-            3 => {
-                self.app.map(|app| {
-                    core::mem::swap(&mut app.slave_rx_buffer, &mut slice);
-                });
-                Ok(slice)
-            }
-            _ => Err((slice, ErrorCode::NOSUPPORT)),
+        // Check if this non-virtualized driver is already in use by
```

Comment:
The other drivers seem to only do the reservation checks on a successful `command` invocation. Do we want to do this everywhere (i.e. subscribe, allow_read{only,write}). This _might_ be beneficial in that we're not necessarily entering the process' Grant region (and thus allocating its Grant region) when we're refusing the operation.

If so, we might want to be consistent and update the other changed drivers (#2518, #2521) as well.

Reply:
Good catch here as well. Based on the discussion on the core call, I removed those extra checks from this driver.